### DISCUSSION TITLE
Attempt to set cookie inside server handle throws TypeError: immutable

### DISCUSSION THREAD

**Discussion by astev89** [sveltejs/kit#6755] [Q&A] [answered]
Example code:

```
        event.cookies.set('test', JSON.stringify(new Date()), {
            path: '/',
            httpOnly: true,
            secure: process.env.NODE_ENV === 'production',
            sameSite: 'strict',
            maxAge: sessionData.exp,
        });

        const response = await resolve(event);
        return response;
```

I'm not sure if the header guard being set to immutable prior to event resolution inside handle is expected or not. Or even if anyone else has experienced this? Currently running `next.480`.

**Comment by teemingc** [comment, marked as answer]
I think you should create an issue for this. This should be clarified in the docs or removed/fixed if it unintended behaviour.

Previously, to set cookies in `handle`, you'd have to add it to `response.headers` after `resolve(event)`.

**Reply by astev89** [reply]
Thank you for the reply and noted. Worth mentioning that using the previous method `response.headers.append('Set-Cookie', ...)` after event resolution is also throwing the same error for me. I'll work on getting a demo repo put together for this.
